### PR TITLE
feat(config): Log possible misconfiguration

### PR DESCRIPTION
**Pull request by najeal**
This PR has originally been opened on Celestia fork of cometbft to resolve this issue : https://github.com/celestiaorg/celestia-app/issues/3736

This adds warn logs when state sync RPCServers are set while enable = false as it may lead to unexpected behavior whereas Config is valid.

#### PR checklist

- [X] Tests written/updated
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [ ] ~~Updated relevant documentation (`docs/` or `spec/`) and code comments~~
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Review by melekes — APPROVED** [review]
Love this solution ❤️ 

I'm not sure we need a new log level, but I'm okay with adding it.

**Comment by najeal** [comment]
The possible misconfiguration can be logged as an Info and I delete the changes in the log package if you prefer 👍

**Comment by najeal** [comment]
Warn log has been deleted

**Review by melekes — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -199,6 +199,16 @@ func (cfg *Config) CheckDeprecated() []string {
 	return warnings
 }
 
+// PossibleMisconfigurations returns a list of possible conflicting entries that
+// may lead to unexpected behavior.
+func (cfg *Config) PossibleMisconfigurations() []string {
+	res := []string{}
+	for _, elem := range cfg.StateSync.PossibleMisconfigurations() {
+		res = append(res, "[statesync] section: %s"+elem)
```
Comment (melekes) [inline config/config.go:207]:
```suggestion
		res = append(res, "[statesync] section: "+elem)
```

**Review by melekes — CHANGES_REQUESTED** [review]

**Review by melekes — APPROVED** [review]

**Pull request merged**